### PR TITLE
Fix unread badges and invite filtering

### LINKED ISSUE #3199: Spaces | Bug Invites are not correctly filtered by space
Currently invites will appear in all spaces instead of just in the one they should belong to.

### LINKED ISSUE #3257: Spaces | Unexpeceted (1) badge with space filter
> it's a DM in the home space (only) 

image

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3257
Fixes #3199

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt [hunk 1]
```diff
@@ -166,24 +166,28 @@ class HomeDetailViewModel @AssistedInject constructor(@Assisted initialState: Ho
                             // TODO!!
                         }
                         is RoomGroupingMethod.BySpace -> {
+                            val activeSpaceRoomId = groupingMethod.spaceSummary?.roomId
                             val dmInvites = session.getRoomSummaries(
                                     roomSummaryQueryParams {
                                         memberships = listOf(Membership.INVITE)
                                         roomCategoryFilter = RoomCategoryFilter.ONLY_DM
+                                        activeSpaceId = activeSpaceRoomId?.let { ActiveSpaceFilter.ActiveSpace(it) } ?: ActiveSpaceFilter.None
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt:174]: `activeSpaceId` should be renamed

File: vector/src/main/java/im/vector/app/features/home/room/list/SpaceRoomListSectionBuilder.kt [hunk 2]
```diff
@@ -129,7 +129,7 @@ class SpaceRoomListSectionBuilder(
                 sections, activeSpaceAwareQueries,
                 R.string.invitations_header,
                 true,
-                RoomListViewModel.SpaceFilterStrategy.NONE
+                RoomListViewModel.SpaceFilterStrategy.NOT_IF_ALL
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/list/SpaceRoomListSectionBuilder.kt:132]: I still do not get the meaning of `NOT_IF_ALL` ha ha

**Review by bmarty — APPROVED** [review]
LGTM, 2 (quite unrelated) remarks

File: vector/src/main/java/im/vector/app/features/home/room/list/SpaceRoomListSectionBuilder.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/home/room/list/SpaceRoomListSectionBuilder.kt:132]: Documented in SpaceFilterStrategy

````
 /**
         * Special case when we don't want to discriminate rooms when current space is null.
         * In this case return all.
         */
        NOT_IF_ALL,
````

File: vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt]: renamed to activeSpaceFilter

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt]: thx

File: vector/src/main/java/im/vector/app/features/home/room/list/SpaceRoomListSectionBuilder.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/list/SpaceRoomListSectionBuilder.kt:132]: Yeah I've seen the comment, thanks, but it's still hard to understand to me ha ha

**Pull request merged**